machine.go

```diff
@@ -362,7 +362,20 @@ func (m *Machine) writerKernelModules(w *writerhelper.WriterHelper) error {
 		"modules.builtin.bin",
 		"modules.devname"}
 
+	usrpath := "/lib/modules"
+	if mergedUsrSystem() {
+		usrpath = "/usr/lib/modules"
+	}
+	data, err := ioutil.ReadFile(path.Join(usrpath, kernelRelease, "modules.builtin"))
+	if err != nil {
+		return err
+	}
+	builtins := string(data)
+
 	for _, v := range modules {
+		if strings.Contains(builtins, v) {
+			continue
+		}
 		usrpath := "/lib/modules"
 		if mergedUsrSystem() {
```

Comment:
Dont you want to remove the `usrpath := ...` snippet here, since you moved it above already?

Reply:
@fdanis-oss ^